### PR TITLE
core/hashes: blake2b256

### PR DESCRIPTION
**Pull request by tynes**
Adds a `Blake2b256Writer` similar to the `Hash256Writer`

### CODE REVIEW

**Comment by tynes** [comment]
I created the test case with the following code:

```javascript
const blake2b = require('bcrypto/lib/blake2b');
const input = Buffer.from('0000', 'hex');
const res = blake2b.digest(input)
console.log(res.toString('hex'))
```

File: core/src/hashes/blake2b256.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+use crate::hashes::marked::MarkedDigestWriter;
+use blake2_rfc::blake2b::Blake2b;
+use std::io::{Result as IOResult, Write};
+
+struct Blake2b256 {
+    ctx: Blake2b,
```
Comment (prestwich) [inline core/src/hashes/blake2b256.rs:6]: the underlying `Blake2b` struct supports `Write` so I think we can cut out this middle-man struct. and simplify the `Write` implementation below

Comment (tynes) [reply core/src/hashes/blake2b256.rs]: Pushed up a new commit that cuts out the middle man struct, much cleaner now :+1:

File: core/src/hashes/blake2b256.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+/// use riemann_core::hashes::{Blake2b256Writer, MarkedDigestWriter};
+///
+/// # fn main() -> Result<()> {
+/// let mut w = Blake2b256Writer::default();
+/// # let data = [0u8; 32];
+///
+/// // Writing more than once will update the hasher.
+/// w.write(&data)?;
+///
+/// // Call finish to consume the produce the hasher and produce the digest.
```
Comment (prestwich) [inline core/src/hashes/blake2b256.rs:22]: propagated my comment typo  from the other file. mind fixing both here and there? :rofl:

Comment (tynes) [reply core/src/hashes/blake2b256.rs]: Pushed a fix to both comments. `consume the produce` lol

**Pull request merged**